pkg/discovery/clustersetip/clustersetip.go

```diff
@@ -0,0 +1,234 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clustersetip
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/reporter"
+	"github.com/submariner-io/submariner-operator/pkg/cidr"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/util/retry"
+	controllerClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Info struct {
+	Enabled bool
+	cidr.Info
+}
+
+type Config struct {
+	ClusterID        string
+	ClustersetIPCIDR string
+	ClusterSize      uint
+}
+
+func AllocateClustersetIPCIDR(clustersetIPInfo *Info) (string, error) {
+	return cidr.Allocate(&clustersetIPInfo.Info) //nolint:wrapcheck // No need to wrap
+}
+
+func isCIDRPreConfigured(clusterID string, clustersetIPNetworks map[string]*cidr.ClusterInfo) bool {
+	// ClustersetIPCIDR is not pre-configured
+	if clustersetIPNetworks[clusterID] == nil || clustersetIPNetworks[clusterID].CIDRs == nil ||
+		len(clustersetIPNetworks[clusterID].CIDRs) == 0 {
+		return false
+	}
+
+	// ClustersetIPCIDR is pre-configured
+	return true
+}
+
+func ValidateClustersetIPConfiguration(clustersetIPInfo *Info, netconfig Config, status reporter.Interface) (string, error) {
```

Comment:
This doesn't need to be public.

```suggestion
func validateClustersetIPConfiguration(clustersetIPInfo *Info, netconfig Config, status reporter.Interface) (string, error) {
```

Reply:
This gets called from test, so need public.